ValheimPlus/GameClasses/EnvMan.cs

```diff
@@ -6,65 +6,90 @@ namespace ValheimPlus.GameClasses
                 {
-                    ValheimPlusPlugin.logger.LogWarning("EnvMan instance not loaded");
+                    if (__instance)
+                    {
+                        __instance.m_dayLengthSec = (long)Configuration.Current.Time.totalDayTimeInSeconds;
+                    }
+                    else
+                    {
+                        ValheimPlusPlugin.Logger.LogWarning("EnvMan instance not loaded");
+                    }
                 }
             }
         }
 
         /// <summary>
-        /// Hook on EnvMan init to alter total day length
+        /// Hook on EnvMan RescaleDayFraction to modify the time of day we are currently in.
         /// </summary>
-        [HarmonyPatch(typeof(EnvMan), "Awake")]
-        public static class TimeInitHook
+        [HarmonyPatch(typeof(EnvMan), nameof(EnvMan.RescaleDayFraction))]
+        public static class RescaleDayFractionPatch
         {
-            private static void Prefix(ref EnvMan __instance)
+            private static void Postfix(ref EnvMan __instance, ref float __result)
             {
-                SetupDayLength();
+                if (Configuration.Current.Time.IsEnabled && Configuration.Current.Time.forcePartOfDay)
+                {
+                    __result = Configuration.Current.Time.forcePartOfDayTime;
+                } else
+                if (Configuration.Current.Time.IsEnabled && Configuration.Current.Time.nightDurationModifier != 0) {
+                    // Day range is: if 0.25f <= __result <= 0.75f
+                    float nightDurationMultiplier = Helper.applyModifierValue(1.0f, Configuration.Current.Time.nightDurationModifier);
+                    if (nightDurationMultiplier <= float.Epsilon)
+                    {
+                        // there is no night, so cut it out.
+                        __result = 0.25f + __result * 0.5f;
+                    }
+                    else
+                    {
+                        nightDurationMultiplier = 1.0f / nightDurationMultiplier;
+                        if (__result >= 0.5f)
+                        {
+                            float stretchFactor = Mathf.Pow((__result - 0.5f) * 2.0f, nightDurationMultiplier);
+                            __result = 0.5f + 0.5f * stretchFactor;
+                        }
+                        else { 
+                            float stretchFactor = 1.0f - Mathf.Pow(((1.0f - __result) - 0.5f) * 2.0f, nightDurationMultiplier);
+                            __result = 0.5f * stretchFactor;
+                        }
+                    }
+                }
             }
         }
 
-        /// <summary>
-        /// Hook on EnvMan to alter night speed
-        /// </summary>
-        [HarmonyPatch(typeof(EnvMan), "FixedUpdate")]
-        public static class TimeUpdateHook
+
+        [HarmonyPatch(typeof(EnvMan), nameof(EnvMan.GetMorningStartSec))]
+        public static class GetMorningStartSecPatch
```

Comment:
class naming 

```suggestion
        public static class EnvMan_GetMorningStartSec_Patch
```

although this one may be a transpile soon.